cabal-install/src/Distribution/Client/CmdOutdated.hs

```diff
@@ -1,510 +1,199 @@
-          projectConfig
-      deps = userConstraintsToDependencies ucnstrs
-      freezeFile = distProjectFile distDirLayout "freeze"
-  freezeFileExists <- doesFileExist freezeFile
+        else fromFlagOrDefault normal (outdatedVerbosity outdatedFlags)
 
-  unless freezeFileExists $
-    dieWithException verbosity $
-      FreezeFileExistsErr freezeFile
+    outdatedFlags :: OutdatedFlags
+    outdatedFlags = extraFlags flags
 
-  debug verbosity $
-    "Reading the list of dependencies from the new-style freeze file " ++ freezeFile
-  return deps
-
--- | Read the list of dependencies from the package description.
-depsFromPkgDesc :: Verbosity -> Compiler -> Platform -> IO [PackageVersionConstraint]
-depsFromPkgDesc verbosity comp platform = do
-  cwd <- getCurrentDirectory
-  path <- tryFindPackageDesc verbosity cwd
-  gpd <- readGenericPackageDescription verbosity path
-  let cinfo = compilerInfo comp
-      epd =
-        finalizePD
-          mempty
-          (ComponentRequestedSpec True True)
-          (const True)
-          platform
-          cinfo
-          []
-          gpd
-  case epd of
-    Left _ -> dieWithException verbosity FinalizePDFailed
-    Right (pd, _) -> do
-      let bd = allBuildDepends pd
-      debug
-        verbosity
-        "Reading the list of dependencies from the package description"
-      return $ map toPVC bd
+extractPackageVersionConstraints :: [PackageSpecifier (SourcePackage UnresolvedPkgLoc)] -> [PackageVersionConstraint]
+extractPackageVersionConstraints =
+  map toPackageVersionConstraint . concatMap genericPackageDependencies . mapMaybe getGenericPackageDescription
   where
-    toPVC (Dependency pn vr _) = PackageVersionConstraint pn vr
-
--- | Various knobs for customising the behaviour of 'listOutdated'.
-data ListOutdatedSettings = ListOutdatedSettings
-  { listOutdatedIgnorePred :: PackageName -> Bool
-  -- ^ Should this package be ignored?
-  , listOutdatedMinorPred :: PackageName -> Bool
-  -- ^ Should major version bumps be ignored for this package?
-  }
-
--- | Find all outdated dependencies.
-listOutdated
-  :: [PackageVersionConstraint]
-  -> SourcePackageDb
-  -> ListOutdatedSettings
-  -> [(PackageVersionConstraint, Version)]
-listOutdated deps sourceDb (ListOutdatedSettings ignorePred minorPred) =
-  mapMaybe isOutdated $ map simplifyPackageVersionConstraint deps
+    getGenericPackageDescription :: PackageSpecifier (SourcePackage UnresolvedPkgLoc) -> Maybe GenericPackageDescription
+    getGenericPackageDescription ps =
+      case ps of
+        NamedPackage{} -> Nothing
+        SpecificSourcePackage x -> Just $ srcpkgDescription x
+
+    toPackageVersionConstraint :: Dependency -> PackageVersionConstraint
+    toPackageVersionConstraint (Dependency name versionRange _) =
+      PackageVersionConstraint name (simplifyVersionRange versionRange)
+
+genericPackageDependencies :: GenericPackageDescription -> [Dependency]
```

Comment:
I wonder if we can preserve the structure of the conditionals when presenting the results to the user. E.g. we can map `CondTree ConfVar [Dependency] component` into `
`CondTree ConfVar [OurResultForDependency] Library` and use that to form the final report.